### PR TITLE
Minor improvement to decorators and AutoBuild

### PR DESCRIPTION
**Pull request by st--**
- makes sure that the AutoBuild metaclass does not overwrite the `__init__`'s docstring anymore
- makes the naming of the functions inside the decorators more consistent
- improves the name_scope() docstring

### CODE REVIEW

File: gpflow/decors.py [hunk 1]
```diff
@@ -29,19 +29,24 @@
...
-    by either `name` option or `name` option is None then name of the
-    function will be used.
+    This decorator wraps a function so that it runs inside a TensorFlow
+    name scope. The name is given by the `name` option; if this is None,
+    then the name of the function will be used.
+    >>> @name_scope()
+    >>> def foo(...):
+    >>>     # now runs inside scope "foo"
+    >>> @name_scope('bar')
+    >>> def @baz(...):
```
Comment (jameshensman) [inline gpflow/decors.py]: is this @ meant to be here?

**Comment by st--** [comment]
Can this be merged now please? Ta:)

File: gpflow/decors.py [hunk 2]
```diff
@@ -29,19 +29,24 @@
...
 def name_scope(name=None):
     """
-    Name scope decorator does little trick with scope naming. The wrapped
-    function will be run inside TensorFlow name scope with name specified
-    by either `name` option or `name` option is None then name of the
-    function will be used.
+    This decorator wraps a function so that it runs inside a TensorFlow
+    name scope. The name is given by the `name` option; if this is None,
+    then the name of the function will be used.
+    >>> @name_scope()
```
Comment (awav) [inline gpflow/decors.py]: Doc generator might use markdown, could you wrap code with *``` ```*

File: gpflow/core/compilable.py [hunk 3]
```diff
@@ -89,6 +89,8 @@ def __init__(self, *args, **kwargs):
             if autobuild_on and global_autobuild_on:
                 self.build()
                 self.initialize(force=True)
+        if origin_init.__doc__:
+            __init__.__doc__ = '\n'.join([origin_init.__doc__, __init__.__doc__])
```
Comment (awav) [inline gpflow/core/compilable.py]: Guess, replacing will be just fine. You don't really need docstring of autobuild in descendant.

Comment (st--) [reply gpflow/core/compilable.py]: Well, the autobuild parses the kwargs before passing them on, so might be useful to keep that in the docstring, no?

Comment (awav) [reply gpflow/core/compilable.py]: autobuild doesn't have any docstrings

Comment (st--) [reply gpflow/core/compilable.py]: It does: core/compilable.py:73: ```            """
            The `kwargs` may or may not contain 'autobuild' option. This option is
            inherited implicitly by all classes.
            """
```

Comment (st--) [reply gpflow/core/compilable.py]: That's the thing that was overwriting the Model `__init__` docstrings in the first place.

**Pull request merged**